lib/salus/scanners/osv/maven_osv.rb

```diff
@@ -0,0 +1,152 @@
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      # Fetch vulnerable dependencies.
+      # Dedupe and select Github Advisory over other sources if available.
+      results = []
+      grouped = match_vulnerable_dependencies(dependencies).group_by { |d| d[:ID] }
+      grouped.each do |_key, values|
+        vuln = {}
+        values.each do |v|
+          vuln = v if v[:Database] == GITHUB_DATABASE_STRING
+        end
+        results.append(vuln.empty? ? values[0] : vuln)
+      end
+
+      # Report scanner status
+      return report_success if results.empty?
+
+      report_failure
+      log(JSON.pretty_generate(results))
+    end
+
+    # Match if dependency version found is in the range of
+    # vulnerable dependency found.
+    def version_matching(version, introduced, fixed)
+      vulnerable_flag = false
+      version_found = SemVersion.new(version)
+
+      if introduced.present? && fixed.present?
+        introduced_version = SemVersion.new(introduced)
+        fixed_version = SemVersion.new(fixed)
+        if version_found >= introduced_version && version_found < fixed_version
+          vulnerable_flag = true
+        end
+      elsif introduced.present?
+        introduced_version = SemVersion.new(introduced)
+        vulnerable_flag = true if version_found >= introduced_version
+      end
+
+      vulnerable_flag
+    end
+
+    # Compare vulnerabilities found with dependencies found
+    # and return vulnerable dependencies
+    def match_vulnerable_dependencies(dependencies)
+      results = []
+      dependencies.each do |dependency|
+        lib = "#{dependency.fetch('group_id', '')}:#{dependency.fetch('artifact_id', '')}"
+        unless dependency['version'].nil?
+          version = dependency['version']
+          # If version is of the format '${deps.version}', log it.
+          if version.match(/\${(.*)\.version}/)
+            bugsnag_notify("MavenOSV: Found #{lib}:#{version} with incompatible format.")
+            next
+          end
+
+          package_matches = @osv_vulnerabilities.select do |v|
+            v.dig("package", "name") == lib
+          end
+          package_matches.each do |m|
+            m["ranges"].each do |version_ranges|
+              introduced = fixed = ""
+              if version_ranges["events"].length == 1
+                if version_ranges["events"][0].key?("introduced")
+                  introduced = version_ranges["events"][0]["introduced"]
+                end
+              elsif version_ranges["events"].length == 2
+                if version_ranges["events"][0].key?("introduced") &&
+                    version_ranges["events"][1].key?("fixed")
+                  introduced = version_ranges["events"][0]["introduced"]
+                  fixed = version_ranges["events"][1]["fixed"]
+                end
+              end
+
+              if version_ranges["type"] == "SEMVER" || version_ranges["type"] == "ECOSYSTEM"
```

Comment:
`update_vul_details_for(results, package_match, introduced, fixed, version_range)`

```
    def update_vul_details_for(results, package_match, introduced, fixed, version_range)
      return unless valid_varsion_range?(version_range)
      return unless version_matching(version, introduced, fixed)

      results.append({
        "Package": package_match.dig("package", "name"),
        "Vulnerable Version": introduced,
        "Version Detected": version,
        "Patched Version": fixed,
        "ID": package_match.fetch("aliases", [package_match.fetch("id", [])])[0],
        "Database": package_match.fetch("database"),
        "Summary": package_match.fetch("summary", package_match.dig("details")).strip,
        "References": package_match.fetch("references", []).collect do |p|
          p["url"]
        end.join(", "),
        "Source":  package_match.dig("database_specific", "url") || DEFAULT_SOURCE,
        "Severity": package_match.dig("database_specific", "severity") || DEFAULT_SEVERITY
      })
    end

    def valid_varsion_range?(version_range)
      version_range["type"] == "SEMVER" || version_range["type"] == "ECOSYSTEM"
    end
```

Reply:
Made changes to it so should be much more readable now.